Weekly Changelog 2023-04-07 to 2023-04-14 (#64802)
CleverRaven/Cataclysm-DDA: data/changelog.txt

## Features:
Tilesets: Automatically prevent occlusions via semi-transparent sprites (similar to retraction)
Add parametric mapgen to consolidate similar basecamp definitions

> Monsters with guntypes will now fire at moving vehicles.
> Enable auto pulping acidic zombies if acid protected
> Add trap_avoids Json property to Monsters to control trap trigger
> Refresh equipment spell duration on recast
> Activities can fire EOCs
> Added full mouse support to the keybindings screen
> Item summoning spells now respect keep hands free option

Reviewer: ```suggestion
Activities can fire EOCs
Added full mouse support to the keybindings screen
```

Add new boss for physics lab
Add 4 New Apartment Complexes
Add several new bungalows

> Correct toad in hole recipe to use toast as well
> Feral Cops
> Add appliances for refilling oxygen tanks
> Removes SPIW Flechette Guns
> Drug lab roof
> Addition of Achievements for major locations
> Fix some incorrect line endings in robotfindskitten
> Motel mapgen update
> Electronic keypad locks

Reviewer 2: ```suggestion
Add appliances for refilling oxygen tanks
Removes SPIW Flechette Guns
Add electronic keypad locks for some pre-generated doors
```

## Interface:
Show crafting failure chances in the crafting interface
Show addictions from hobbies in newcharacter tab
Added ability to think to yourself in the message log
Added Presets for pocket settings
Add 'solarized' to color themes
Add fast scrolling support to camp mission UI
If item has two use actions, show both in Use Item menu

> Added force drop option to player/NPC debug menu.

Reviewer 2: ```suggestion
```

Add new boss for physics lab
Add 4 New Apartment Complexes
Add several new bungalows

> different blood types for acidic monsters
> large bones
> Makes Repeating Crossbow not a SMG
> map update: home improvement superstore
> Map update: lumbermill & lumberyard
> map update: landscaping supply
> map update: large power substation
> map update: St. John's Farm
> map update: evac shelter
> map update: bar
> map update: dispensary
> Add the first Exodii mission
> map update: craft shop 2
> map update: bowling alley
> Combat Capable Mounts
> New construction: rocky obstacle
> More scenarios, achievements and EOC effects
> Map update: 3x3 hotels
> refugee caravan

Reviewer: ```suggestion
Updates to several existing maps
Add the first Exodii mission
Combat Capable Mounts
```

## Interface:
Show crafting failure chances in the crafting interface
Show addictions from hobbies in newcharacter tab
Added ability to think to yourself in the message log
Added Presets for pocket settings

> New upgrades to spacebar sidebar.
> Prettier loading UI for loading the save
> Player display speed tab effect list is more comprehensive
> Add min/max feedback to new character stats interface

Reviewer: ```suggestion
Prettier loading UI for loading the save
Add min/max feedback to new character stats interface
```

[Magiclysm Graphical Overmap] Delete Magiclysm Graphical Overmap
[Magiclysm] Another approach to animist summoning
[Xedra Evolved] Add content for dreamsmiths and dreamers
[Bombastic Perks] Add a new mod where the player unlocks perks as they progress through the game

> [Innawood] Adds back forest trails. Also no longer requires city size to be set to 0.

Reviewer 3: ```suggestion
```

[Magiclysm Graphical Overmap] Delete Magiclysm Graphical Overmap
[Magiclysm] Another approach to animist summoning
[Xedra Evolved] Add content for dreamsmiths and dreamers

> Adds tamablity to newer birds in Tamable_Wildlife Mod
> New mod where the player unlocks perks
> [Innwood] Adds back forest trails. Also no longer requires city size to be set to 0.
> Added documentation and configuration menu for Bombastic Perks
> More Bombastic Perks

Reviewer 2: ```suggestion
[Bombastic Perks] Add a new mod where the player unlocks perks as they progress through the game
[Innawood] Adds back forest trails. Also no longer requires city size to be set to 0.
```

Removed scent tracking from certain zombies
Cody can make chainmail armor and charges more
Rework melee, unarmed, dodge, cutting, stabbing, and bashing practice recipes to limit higher level practice actions to books

> Salisfy and dahlia can be bulk-cooked. Salsify is now safe to eat in any form.
> improve protective lenses CBM
> Chargen description proofreading and a few changes to profession loadouts
> Gives the sugar beet syrup recipe a batch crafting time

Reviewer 2: ```suggestion
```

[Magiclysm Graphical Overmap] Delete Magiclysm Graphical Overmap
[Magiclysm] Another approach to animist summoning
[Xedra Evolved] Add content for dreamsmiths and dreamers

> adds a configuration menu for perks
> [Bombastic Perks] More perks
> [DinoMod] integrated mutation armor
> [DinoMod] therizinosaurus

Reviewer: ```suggestion
[Bombastic Perks] New mod to earn perks through gameplay
[DinoMod] Integrated mutation armor
[DinoMod] Therizinosaurus
```

Flush map buffers after failing to create starting location
Smart controller supports using only one engine
Disassembly doesn't return items with UNRECOVERABLE flag
Fix diary text navigation wrapping
Fix display of diagonal vehicle glyphs

> Eliminate unnecessary long road connections

Reviewer 3: ```suggestion
```

Reviewer 2: I think changing this might deserve to be in the changelog, but I also couldn't really tell how impactful it was.

Removed scent tracking from certain zombies
Cody can make chainmail armor and charges more
Rework melee, unarmed, dodge, cutting, stabbing, and bashing practice recipes to limit higher level practice actions to books

> Epicuticle has worse encumbrance, defense and breathability
> Adjusted volume and weight stats for motors/engines to be more realistic
> Make makeshift bandages not require tailoring, making them easier to craft at 0 tailoring
> Remove flaming eye phantom melee attack that can disrupt aim.
> Converts most of the armor and clothing still using the old limb system to the new system with sub-limbs, part 3: Glorious gloves and snazzy smallclothes
> Phase Immersion Armor has an inbuilt battery
> Add bundling for synethic fabric
> Adjusting recipes using electric motors to use correct motor size relative to resulting item sizes
> Changes fur and faux fur tailoring to require Fabric Cutting 1 instead of 2
> Bread flour can now also be made with a food processor
> Ye Scots Beuk o Cuikery can teach ye how ta make sausage casings

Reviewer: ```suggestion
Remove flaming eye phantom melee attack that can disrupt aim
Converts most of the armor and clothing still using the old limb system to the new system with sub-limbs
```

Flush map buffers after failing to create starting location
Smart controller supports using only one engine
Disassembly doesn't return items with UNRECOVERABLE flag

> Fix diary text navigation wrapping
> Avoid spilling pocket contents into the origin item.
> Add unique feline mutations to Extra Mutated Scenarios mod
> Adds a roof to generic mines
> Adds a new itemgroup to spawn the paint in the tutorial in cans instead of spilled
> Fixed malicious mane having 0 weight or volume
> Fixing an infinite loop when unloading a container that contains a spillable container
> Diagonal vehicle glyph correction
> Eliminate unnecessary long road connections

Reviewer 2: ```suggestion
Fix diary text navigation wrapping
Fix display of diagonal vehicle glyphs
Eliminate unnecessary long road connections
```

Flush map buffers after failing to create starting location
Smart controller supports using only one engine
Disassembly doesn't return items with UNRECOVERABLE flag

> Make elephant gun a double rifle to match reality
> feral cops become zombie cops
> mobile home park: no more indoor flagpoles
> Fixed regression in inventory screens where pressing an item's hotkey does not work as expected
> Fix pipebomb length
> Fix wall cling phasing through floors
> Fixes the name shown in the world creation keybinding menu for the mod manager.
> Exeption in tutorial (debug mode) while using command 'a'
> Add ability to remove plants from planters without destroying planter
> fixes #64758 by removing enemy spawns from the lighthouse, and removing the lake cabin from the safe place scenario.
> FIxing nomad armor functionality
> Crash on achievement gain
> Prevent broken vp in-place replacement when racked
> Do not set charges to -1 on items that are not count by charges when filling a pocket
> Wield Wiedget hides (Folded) and tname now includes default_mods
> fix Energy consumption of 'Taste Modifier CBM' when consuming food with positive enjoyment
> Spellcasting tools no longer waste charges if you cancel out and don't actually cast the spell
> Fix erroneous limb modifier
> Fix message about rot when eating with saprophage 
> Fix using unload_everything zone to remove gunmod gets copies of gunmod.
> Ferals can use their guns in GG
> Hoists now provide ROPE 2 and SUSPENDING 2 qualities

Reviewer: ```suggestion
Feral cops become zombie cops
Fix wall cling phasing through floors
Add ability to remove plants from planters without destroying planter
Prevent broken vp in-place replacement when racked
Spellcasting tools no longer waste charges if you cancel out and don't actually cast the spell
Fix using unload_everything zone to remove gunmod gets copies of gunmod
Ferals can use their guns in GG
```

Epower and power in units::power
Update compiler support.  Now supporting gcc 8.1+, clang 10+, XCode 10.1+
Refactor some crafting infrastructure to support removal of charges
Unify gun battery/ups/bionic energy consumption
Modernize string_formatter
Migrate some JSON APIs to string_view

## Build:
Support Mac arm64 build

## I18N and A11Y:

> More accurately describe Uyen Tran's task requirements
> Add percentage-translated statistic to language selection

Reviewer: ```suggestion
Add percentage-translated statistic to language selection
```
